### PR TITLE
Handle tag/commit type case in github

### LINKED ISSUE #40: 404 Not Found when fetching BuildInfo from some WildFly repositories
Fetching BuildInfo data from some WildFly repositories doesn't work and returns `Not Found, status code 404`.
Examples of tags that don't work:
https://github.com/wildfly/wildfly/releases/tag/32.0.1.Final
https://github.com/wildfly/wildfly-core/releases/tag/25.0.2.Final
https://github.com/wildfly/quickstart/releases/tag/33.0.2.Final

Weirdly, one of the WildFly tags works:
https://github.com/wildfly/wildfly/releases/tag/33.0.1.Final

Some other repositories also work:
https://github.com/wildfly/wildfly-maven-plugin/releases/tag/5.0.1.Final
https://github.com/wildfly/wildfly-arquillian/releases/tag/5.0.0.Final

### PR DESCRIPTION
**Pull request by petrberan**
Resolves #40

### CODE REVIEW

**Review by khermano — CHANGES_REQUESTED** [review]
Looks good, but we should also take care about the 404. When the tag format is valid but tag does not exist (e.g.: https://github.com/wildfly/wildfly-arquillian/releases/tag/8.0.1.Final).

**Comment by petrberan** [comment]
Added exception handling @khermano

File: src/main/java/org/jboss/set/exception/GitRestClientExceptionMapper.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.ext.ExceptionMapper;
+import jakarta.ws.rs.ext.Provider;
+import org.jboss.logging.Logger;
+import org.jboss.resteasy.reactive.ClientWebApplicationException;
+import org.jboss.set.GitBuildInfoAssembler;
+
+@Provider
+public class GitRestClientExceptionMapper implements ExceptionMapper<ClientWebApplicationException> {
```
[the remark this answers is not in the document]
Comment (petrberan) [reply src/main/java/org/jboss/set/exception/GitRestClientExceptionMapper.java:11]: Agree, this is was just the fastest solution.

Feel free to open an issue

**Comment by petrberan** [comment]
Changed some of the lines there.

I'll leave it at this state, the response with some user-friendly lines are needed across the project and since this is just a warning in the logger we might as well leave it for the next beta release.

Can you create an issue to provide more robust Responses with some valid messages @khermano ?

**Review by khermano — APPROVED** [review]
LGTM, I will create mentioned issues. Thanks @petrberan

**Pull request merged**